linera-rpc/src/grpc/transport.rs

```diff
@@ -42,6 +43,7 @@ cfg_if::cfg_if! {
             if let Some(timeout) = options.timeout {
                 endpoint = endpoint.timeout(timeout);
             }
+            endpoint = endpoint.keep_alive_timeout(Duration::from_millis(KEEPALIVE_TIME_MS));
```

Comment:
Why is it not another one option?

Reply:
Just copy past from timeout

Reply:
This is intentionally not an option. The reason is that this _should_ be negotiated on a per-validator basis. The reason is that different ingresses have different timeouts.

However after further discussion with @Twey , we think a better implementation may be to have the keepalive originate from the [server](https://docs.rs/tonic/latest/tonic/transport/struct.Server.html#method.http2_keepalive_interval) - hence why this is still a draft.

The server-side option _will_ be configurable.